RoddyCore/src/de/dkfz/roddy/config/RecursiveOverridableMapContainer.java

```diff
@@ -105,17 +104,53 @@ public boolean hasValue(String id) {
         return getListOfAllValues().contains(id);
     }
 
+    /**
+     * When you have several configuration objects, e.g.
+     * A -> B
+     *
+     * A contains values a and b, where a == abc and b has a reference to a b=${a}
+     * B contains values a'  == def
+     * If you query B for b, b holds a reference to configuration A
+     * Therefore, b.toString() will result in b == abc and NOT in def as expected.
+     *
+     * By elevating a copy of b to B, everything will be fixed, as B can be the parent of b'
+     *
+     * This is true for ConfigurationValue, which rely on dependencies between themselves.
+     *
+     * In RecursiveOverridableMapContainerForConfigurationValues, we override the method and
+     * copy the value and set the configuration reference to (in this example) B
+     *
+     * @param src
+     * @return
+     */
+    protected V copyValueToThisContainer(V src) {
+        add(src);
+        return src;
+    }
+
+    /**
+     * Does not check, if the value exists!
+     * @param id
+     * @return
+     */
+    protected V _getValueUnchecked(String id) {
+        V cval = getAllValues().get(id);
+        if(!values.containsKey(id))
+            return copyValueToThisContainer(cval);
```

Comment:
Hm, this adds a side-effect to a get-operation. I'm not happy with this. Given our discussion, would be querying the full tree be an alternative?